### PR TITLE
[TASK] Update the code of conduct to version 2.1

### PR DESCRIPTION
**Pull request by oliverklee**
Copy version 2.1 of the contributor convenant code of conduct without any changes except those:

- use our contact email address
- drop links additional resources related to the contributor covenant CoC

https://www.contributor-covenant.org/version/2/1/code_of_conduct/

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,76 +1,119 @@
...
 ## Attribution
 
 This Code of Conduct is adapted from the [Contributor Covenant][homepage],
-version 1.4, available at
-[http://contributor-covenant.org/version/1/4/][version].
-
-[homepage]: http://contributor-covenant.org
-[version]: http://contributor-covenant.org/version/1/4/
+version 2.1, available at
+[https://www.contributor-covenant.org/version/2/1/code_of_conduct.html][v2.1].
```
Comment (JakeQZ) [inline CODE_OF_CONDUCT.md:119]: The link doesn't work.  I think we need to lose the square brackets, and the version number at the end (the version is already in the URL).
```suggestion
https://www.contributor-covenant.org/version/2/1/code_of_conduct.html.
```

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -1,76 +1,119 @@
...
+**Community Impact**: Demonstrating a pattern of violation of community
+standards, including sustained inappropriate behavior, harassment of an
+individual, or aggression toward or disparagement of classes of individuals.
+
+**Consequence**: A permanent ban from any sort of public interaction within the
+community.
 
 ## Attribution
 
 This Code of Conduct is adapted from the [Contributor Covenant][homepage],
```
Comment (JakeQZ) [inline CODE_OF_CONDUCT.md:117]: I think this invalid markdown should be fixed as part of this change.
```suggestion
This Code of Conduct is adapted from the [Contributor Covenant](https://www.contributor-covenant.org/),
```

**Review by JakeQZ — CHANGES_REQUESTED** [review]
All seems fine, apart from invalid markdown and broken links in the attribution section.

**Review by jjriv — APPROVED** [review]

**Review by JakeQZ — APPROVED** [review]
Looks fine now.

**Pull request merged**